worker/uploader.go

```diff
@@ -126,10 +129,17 @@ outer:
 			}
 
 			// track stats
-			success, failure, uploadEstimateMS, uploadSpeedBytesPerMS := handleSectorUpload(err, duration, time.Since(start), req.overdrive, u.logger)
+			success, failure, uploadEstimateMS, uploadSpeedBytesPerMS := handleSectorUpload(err, duration, elapsed, req.overdrive)
 			u.trackSectorUploadStats(uploadEstimateMS, uploadSpeedBytesPerMS)
 			u.trackConsecutiveFailures(success, failure)
 
+			// debug log
+			if uploadEstimateMS > 0 && !success {
```

Comment:
Isn't it enough to check for `uploadEstimateMS` greater than zero or equal to distinguish between whether a penalty was applied vs not? Cause the conditions here are not obvious without knowing the inner workings of `handleSectorUpload`.

Reply:
Well no because we overload the `uploadEstimate` with a penalty. In the happy case for instance this value will just be the time it took to upload the sector to that host and we want to track that datapoint, in which case I don't want to log it as a penalty. 

I would generally agree with you and just update the logic but in this case I feel it's simple enough? I don't mind splitting it up though or rework the return variables of `handleSectorUpload`. I don't like the two bools (`success` and `failure` to handle the 3 `consecutiveFailures` cases either). I'll take another stab at it.